### PR TITLE
Setup phpstan and fix reported errors

### PR DESCRIPTION
**Pull request by pyrech**
- Adds phpstan
- Fix to level 6
- Fix deprecation in phpunit
- Add phpstan to the continuous integration checks

### CODE REVIEW

File: src/Statistic/StatisticCollector.php [hunk 1]
```diff
@@ -18,12 +18,15 @@ class StatisticCollector
 {
     private $client;
 
+    /**
+     * @param Client<string, Client> $client
```
Comment (damienalexandre) [inline src/Statistic/StatisticCollector.php:22]: I don't understand this phpdoc, this argument is not an array

File: tests/Controller/ContentControllerTest.php [hunk 2]
```diff
@@ -15,7 +15,7 @@
 
 class ContentControllerTest extends WebTestCase
 {
-    public function test_homepage_works()
+    public function test_homepage_works(): void
```
Comment (damienalexandre) [inline tests/Controller/ContentControllerTest.php:18]: Not a big fan of typing not critical code / test code that never ever ever return anything (as it's a test).

File: src/Statistic/StatisticCollector.php (hunk 1 above)
Comment (pyrech) [reply src/Statistic/StatisticCollector.php:22]: Client is an IteratorAggregate returning an ArrayIterator of itself (see https://github.com/predis/predis/blob/main/src/Client.php#L41) thus PHPStan requires us to specify the type iterated

File: tests/Controller/ContentControllerTest.php (hunk 2 above)
Comment (pyrech) [reply tests/Controller/ContentControllerTest.php:18]: Not a big fan either but I think it do not harm us to be forced to write typed PHP :grin:

**Review by damienalexandre — APPROVED** [review]
Looks good, let's wait for the CI to run & merge :)

**Pull request merged**